pkgs/applications/blockchains/ton/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, git
+, gperf
+, libmicrohttpd
+, openssl
+, readline
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ton";
+  version = "2022.12";
+
+  src = fetchFromGitHub {
+    owner = "ton-blockchain";
+    repo = "ton";
+    rev = "v${version}";
+    sha256 = "sha256-/eY/rPHQEoPjdbHmTpg2f8rjkg8tKAed4oF8KoMqz8k=";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    # do not set TON_ARCH or it will use wrong -mcpu/-march
+    substituteInPlace CMakeLists.txt \
+      --replace 'set(TON_ARCH "native" CACHE STRING "Architecture, will be passed to -march=")' "" \
+      --replace 'set(TON_ARCH "apple-m1")' ""
+  '';
```

Comment:
Isn't our cc-wrapper filter these flags out?

Reply:
Yes, but not "apple-m1". Fixed.